lib/main.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:appflowy_theme_marketplace/src/authentication/data/repositories/firebase_authentication_repository.dart';
+import 'package:appflowy_theme_marketplace/src/payment/application/payment_bloc/payment_bloc.dart';
+import 'package:appflowy_theme_marketplace/src/payment/data/repositories/stripe_payment_repository.dart';
+import 'package:appflowy_theme_marketplace/src/plugins/domain/repositories/plugin_repository.dart';
+import 'package:appflowy_theme_marketplace/src/plugins/domain/repositories/ratings_repository.dart';
+import 'package:appflowy_theme_marketplace/src/user/application/bloc/user_bloc.dart';
+import 'package:appflowy_theme_marketplace/src/user/data/firebase_repository/firebase_user_repository.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:get_it/get_it.dart';
+import 'src/authentication/application/auth_bloc/auth_bloc.dart';
+import 'src/authentication/presentation/signin_page/signin.dart';
+import 'src/plugins/application/plugin/plugin_bloc.dart';
+import 'src/plugins/data/firebase_repository/firebase_plugin_repository.dart';
+import 'src/plugins/data/firebase_repository/firebase_rating_repository.dart';
+import 'src/plugins/presentation/dashboard_page/dashboard.dart';
+import 'src/user/domain/repositories/user_repository.dart';
+import 'src/user/presentation/register_page/register.dart';
+
+final getIt = GetIt.instance;
+
+Future<void> main() async {
+  getIt.registerSingleton<PluginRepository>(FirebasePluginRepository());
+  getIt.registerSingleton<RatingsRepository>(FirebaseRatingsRepository());
+  getIt.registerSingleton<UserRepository>(FirebaseUserRepository());
+  getIt.registerSingleton<FirebaseAuthenticationRepository>(FirebaseAuthenticationRepository());
+  getIt.registerSingleton<StripePaymentRepository>(StripePaymentRepository());
+
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
+
+  runApp(
+    MultiBlocProvider(
+      providers: [
+        BlocProvider<AuthBloc>(
+          create: (BuildContext context) => AuthBloc(
+            authenticationRepository:
+                getIt.get<FirebaseAuthenticationRepository>(),
+          ),
+        ),
+        BlocProvider<PluginBloc>(
+          create: (BuildContext context) => PluginBloc(
+            pluginRepository: getIt.get<PluginRepository>(),
+            ratingsRepository: getIt.get<RatingsRepository>(),
+          ),
+        ),
+        BlocProvider<UserBloc>(
+          create: (BuildContext context) => UserBloc(
+            userRepository: getIt.get<UserRepository>(),
+          ),
+        ),
+        BlocProvider<PaymentBloc>(
+          create: (BuildContext context) => PaymentBloc(
+              paymentRepository: getIt.get<StripePaymentRepository>()),
+        ),
+      ],
```

Comment:
It would be better to refactor the code inside the `runApp` function into a new widget."